pkg/cmd/release-payload-controller/payload_verification_controller.go

```diff
@@ -0,0 +1,185 @@
+	}()
+
+	if !cache.WaitForNamedCacheSync("PayloadVerificationController", ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go func() {
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *PayloadVerificationController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *PayloadVerificationController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx, key.(string))
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *PayloadVerificationController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting PayloadVerificationController sync")
+	defer klog.V(4).Infof("PayloadVerificationController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	if err != nil {
+		// The ReleasePayload resource may no longer exist, in which case we stop processing.
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	releasePayload := originalReleasePayload.DeepCopy()
+
+	// If there are any JobResults defined, then we don't need to do anything else here...
+	if len(releasePayload.Status.AnalysisJobsResults) != 0 ||
+		len(releasePayload.Status.BlockingJobsResults) != 0 ||
+		len(releasePayload.Status.InformingJobsResults) != 0 {
+		klog.V(4).Infof("ReleasePayload: '%s/%s' already synced", releasePayload.Namespace, releasePayload.Name)
+		return nil
+	}
+
+	klog.V(4).Infof("Syncing Payload Verification for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	for name, verify := range releasePayload.Spec.PayloadVerification {
+		JobStatus := v1alpha1.JobStatus{
+			JobName:    verify.ProwJob.Name,
+			JobAlias:   name,
+			MaxRetries: verify.MaxRetries,
+			Optional:   verify.Optional,
+		}
+
+		if verify.AggregatedProwJob != nil {
+			releasePayload.Status.AnalysisJobsResults = append(releasePayload.Status.AnalysisJobsResults, JobStatus)
+		} else if !verify.Optional {
+			releasePayload.Status.BlockingJobsResults = append(releasePayload.Status.BlockingJobsResults, JobStatus)
+		} else {
+			releasePayload.Status.InformingJobsResults = append(releasePayload.Status.InformingJobsResults, JobStatus)
+		}
+	}
+
+	if !reflect.DeepEqual(originalReleasePayload, releasePayload) {
+		_, err := c.releasePayloadClient.ReleasePayloads(releasePayload.Namespace).Update(ctx, releasePayload, v1.UpdateOptions{})
```

Comment:
This should be `UpdateStatus`, if you don't have UpdateStatus it means your CRD definition is missing the 'enable status subresource" bit set and you need to turn it on.

Reply:
Done